roles/icinga_plugins/molecule/default/converge.yml

```diff
@@ -0,0 +1,159 @@
+  hosts: all
+
+  pre_tasks:
+    - name: install epel repo
+      yum:
+        name: epel-release
+        state: installed
+      when: ansible_os_family == 'RedHat'
+      tags:
+        - molecule-idempotence-notest
+
+    - name: import icinga2 key
+      command: "rpm --import https://packages.icinga.com/icinga.key"
+      when: ansible_os_family == 'RedHat'
+      tags:
+        - molecule-idempotence-notest
+
+    - name: install icinga2 repo
+      command: "yum install https://packages.icinga.com/epel/icinga-rpm-release-7-latest.noarch.rpm -y"
+      when:
+        - ansible_os_family == 'RedHat'
+        - ansible_facts.distribution_major_version | int is version('7', '=')
+      tags:
+        - molecule-idempotence-notest
+
+    - name: install icinga2 repo
+      command: "yum install https://packages.icinga.com/epel/icinga-rpm-release-8-latest.noarch.rpm -y"
+      when:
+        - ansible_os_family == 'RedHat'
+        - ansible_facts.distribution_major_version | int is version('8', '=')
+      tags:
+        - molecule-idempotence-notest
+
+    - name: install icinga2 basic packages
+      apt:
+        name: gnupg
+        state: present
+        update_cache: true
+      when: ansible_os_family == 'Debian'
+      tags:
+        - molecule-idempotence-notest
+
+    - name: add icinga2 repo key
+      apt_key:
+        url: "http://packages.icinga.com/icinga.key"
+        state: present
+      when: ansible_os_family == 'Debian'
+      tags:
+        - molecule-idempotence-notest
+
+    - name: add icinga2 repository on debian 11
+      apt_repository:
+        repo: "{{ item }}"
+        state: present
+        update_cache: true
+      with_items:
+        - deb http://packages.icinga.com/debian icinga-bullseye main
+        - deb-src http://packages.icinga.com/debian icinga-bullseye main
+      when:
+        - ansible_os_family == 'Debian'
+        - ansible_facts.distribution_major_version | int is version('11', '=')
+      tags:
+        - molecule-idempotence-notest
+
+    - name: install icinga2 build dependencies
+      apt:
+        pkg: "icinga2"
+        state: "build-dep"
+        update_cache: true
+      when: ansible_os_family == 'Debian'
+      tags:
+        - molecule-idempotence-notest
+
+    - name: install icinga2 basic packages
+      apt:
+        name: "icinga2"
+        state: "present"
+        update_cache: true
+      when: ansible_os_family == 'Debian'
+      tags:
+        - molecule-idempotence-notest
+
+    - name: set name of installation package based on distro
+      set_fact:
+        icinga_install_plugins:
+          - nagios-plugins-all
+      when: ansible_os_family == 'Redhat'
+
+    - name: set name of installation package based on distro
+      set_fact:
+        icinga_install_plugins:
+          - monitoring-plugins
+      when: ansible_os_family == 'Debian'
+
+  collections:
+    - t_systems_mms.ansible_collection_icinga
+
+  roles:
+    - icinga_plugins
+
+  post_tasks:
```

Comment:
These tasks look like they are tests. If it is so, they belong into the `verify`-phase.